crates/move-stackless-bytecode/src/axiom_function_analysis.rs

```diff
@@ -0,0 +1,114 @@
+use codespan_reporting::diagnostic::Severity;
+use move_model::{
+    model::FunctionEnv,
+    ty::{PrimitiveType, Type},
+};
+
+use crate::{
+    deterministic_analysis,
+    function_target::FunctionData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    pure_function_analysis::PureFunctionAnalysisProcessor,
+};
+
+pub struct AxiomFunctionAnalysisProcessor();
+
+impl AxiomFunctionAnalysisProcessor {
+    pub fn new() -> Box<Self> {
+        Box::new(Self())
+    }
+
+    fn check_parameters(&self, func_env: &FunctionEnv) -> bool {
+        for param in func_env.get_parameters() {
+            if param.1.is_reference() {
+                func_env.module_env.env.diag(
+                    Severity::Error,
+                    &func_env.get_loc(),
+                    &format!(
+                        "Axiom functions cannot have reference parameters: '{}'",
+                        func_env.symbol_pool().string(param.0)
+                    ),
+                );
+                return false;
+            }
+        }
+
+        if func_env.get_return_count() != 1 {
+            func_env.module_env.env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Axiom functions must have exactly one return value",
+            );
+            return false;
+        }
+
+        if !matches!(
+            func_env.get_return_types()[0],
+            Type::Primitive(PrimitiveType::Bool)
+        ) {
+            func_env.module_env.env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Axiom functions should return bool type",
+            );
+            return false;
+        }
+
+        if func_env.get_type_parameter_count() != 0 {
+            func_env.module_env.env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Axiom functions cannot have type parameters",
+            );
+            return false;
+        }
```

Comment:
I think type parameters are ok, but then it's a question of what instances should be used